### PR TITLE
Home carousel consuming

### PR DESCRIPTION
**Pull request by CamiloGilFranco**
## Summary

- The content of the main carousel is generated from the consumption of the backend

- The content of the RestaurantFilterList is generated from the consumption of the backend

- It was necessary to modify the filter function to allow the consumption of data from the backend

- Data consumption is generated for RestaurantList

- Rendered content of the Restaurant Card Component is modified by removing the schedule and the minimum price

### CODE REVIEW

File: src/components/RestaurantCardComponent/RestaurantCardComponent.jsx [hunk 1]
```diff
@@ -35,13 +33,23 @@ const RestaurantCardComponent = ({
...
+            </span>
+          ))}
+          {/* <span className="restaurant-card-component-info-list-cuisine">
+            - {categories[0]}
+          </span>
+          <span className="restaurant-card-component-info-list-cuisine">
+            - {categories[1]}
+          </span>
+          <span className="restaurant-card-component-info-list-cuisine">
+            - {categories[2]}
```
Comment (ht1204) [inline src/components/RestaurantCardComponent/RestaurantCardComponent.jsx:51]: What is this commented code doing?

File: src/components/RestaurantList/RestaurantList.jsx [hunk 2]
```diff
@@ -52,21 +72,32 @@ const RestaurantList = () => {
...
         });
       });
     };
 
-    if (rating >= 2 && categories.length === 0)
-      result = data.filter((element) => Math.round(element.rating) >= rating);
-    if (categories.length > 0 && rating < 2) result = filterByCategory(data);
-    if (categories.length > 0 && rating >= 2)
-      result = data.filter(
+    if (rating >= 2 && categoriesArray.length === 0) {
```
Comment (ht1204) [inline src/components/RestaurantList/RestaurantList.jsx]: `if (rating >= 2 && !categoriesArray.length)`

File: src/components/carrusel/CarrucelComponent.jsx [hunk 3]
```diff
@@ -45,21 +54,26 @@ const CarrucelComponent = () => {
     <main className="home-carousel-component">
       <section className="home-carousel-container">
         <Slider {...settings} className="home-carousel-main">
-          {keys.map((element, index) => {
+          {cuisinesList.map((element, index) => {
+            console.log(element);
```
Comment (ht1204) [inline src/components/carrusel/CarrucelComponent.jsx]: What is this console.log doing?

File: src/components/carrusel/CarrucelComponent.jsx [hunk 4]
```diff
@@ -45,21 +54,26 @@ const CarrucelComponent = () => {
     <main className="home-carousel-component">
       <section className="home-carousel-container">
         <Slider {...settings} className="home-carousel-main">
-          {keys.map((element, index) => {
+          {cuisinesList.map((element, index) => {
```
Comment (ht1204) [inline src/components/carrusel/CarrucelComponent.jsx:57]: Why not destructuring the element item object?

Comment (ht1204) [inline src/components/carrusel/CarrucelComponent.jsx:57]: Why not using an UUID or nanoid instead off using index for mapping jsx?

File: src/components/RestaurantList/RestaurantList.jsx [hunk 5]
```diff
@@ -52,21 +72,32 @@ const RestaurantList = () => {
...
+      result = sortList.filter(
+        (element) => Math.round(element.rating) >= rating
+      );
+    }
+    if (categoriesArray.length > 0 && rating < 2) {
```
Comment (ht1204) [inline src/components/RestaurantList/RestaurantList.jsx]: `if (categoriesArray.length && rating < 2)`

File: src/components/RestaurantList/RestaurantList.jsx [hunk 6]
```diff
@@ -52,21 +72,32 @@ const RestaurantList = () => {
...
+      result = filterByCategory(sortList);
+    }
+    if (categoriesArray.length > 0 && rating >= 2) {
```
Comment (ht1204) [inline src/components/RestaurantList/RestaurantList.jsx]: Same here

**Review by ht1204 — CHANGES_REQUESTED** [review]
Consider the comments for changes, thx.

File: src/components/carrusel/CarrucelComponent.jsx (hunk 4 above)
Comment (CamiloGilFranco) [reply src/components/carrusel/CarrucelComponent.jsx:57]: The index is being used in all the maps on Sergio's recommendation, I don't really know the methods you mention

Comment (CamiloGilFranco) [reply src/components/carrusel/CarrucelComponent.jsx:57]: I tried but doesn't work

File: src/components/RestaurantCardComponent/RestaurantCardComponent.jsx (hunk 1 above)
Comment (CamiloGilFranco) [reply src/components/RestaurantCardComponent/RestaurantCardComponent.jsx:51]: My bad, I already have it corrected to upload it in the next PR

File: src/components/carrusel/CarrucelComponent.jsx (hunk 4 above)
Comment (ht1204) [reply src/components/carrusel/CarrucelComponent.jsx:57]: ok, no problem, let's move ahead, thx 4 clarifying.

Comment (ht1204) [reply src/components/carrusel/CarrucelComponent.jsx:57]: or else using the entity id, just for better practices in mapping, let's move then.

**Review by ht1204 — APPROVED** [review]

**Pull request merged**